internal/repos/util.go

```diff
@@ -4,33 +4,29 @@ import (
 	"net/url"
 )
 
-// setUserinfoBestEffort adds the username and password to rawurl. If user is
-// not set in rawurl, username is used. If password is not set and there is a
+// setUserinfoBestEffort adds the username and password to rawurl. If username param
+// is not set, user from url is used. If password is not set and there is a
 // user, password is used. If anything fails, the original rawurl is returned.
```

Comment:
I think the comment could be updated to match the new logic; specifically that so long as `username` argument is non empty then we will skip using the UserInfo associated with the `rawurl` . Feel free to modify the suggestion below and I'll review when I get a chance

```suggestion
// setUserinfoBestEffort updates the url to utilize the username param and also utilize
// the password param if provided. If username param is not provided then
// utilize the UserInfo from rawurl to determine username and password to use for the url.
```

Reply:
The other clarification I was hoping to add in that suggestion is that password param could be ignored even if non-empty/non-nil, if the username param is empty.